cosmos/operators/local.py

```diff
@@ -464,7 +477,8 @@ def _handle_datasets(self, context: Context) -> None:
         outlets = self.get_datasets("outputs")
         self.log.info("Inlets: %s", inlets)
         self.log.info("Outlets: %s", outlets)
-        self.register_dataset(inlets, outlets, context)
+        if AIRFLOW_VERSION < _AIRFLOW3_VERSION:
```

Comment:
I feel it maybe better to introduce these conditions in the respective method and based on the Airflow version, for now we could decide on whether to call the inner function or not. Later on, we could call Airflow version specific inner functions within those functions to handle both versions. WDYT?

Reply:
I thought it, but I found this approach easier for handling decorator calls like provide_session. However, I'll give it some more thought and, if possible, will create a follow-up PR.